app/controllers/invitations_global_controller.rb

```diff
@@ -0,0 +1,45 @@
+class InvitationsGlobalController < ApplicationController
```

Comment:
@shir I shared these models because they have different validation, different behavior of the same methods, I was trying to avoid a large number of `if` and `case` states. This is my main aim.